src/ocl_utils_mod.f90

```diff
@@ -46,16 +57,31 @@ subroutine init_device(device, version_str, context)
                             num_platforms)
     call check_status('clGetPlatformIDs', ierr)
 
-    ! Get device IDs only for platform 1
-    iplatform=1
+    ! Select the OpenCL platform to use
+    CALL get_environment_variable("FORTCL_PLATFORM", strvalue, status=ierr)
+    if(ierr .eq. 1) then
+        ! By default use platform 1
+        iplatform = 1
+    else
+        read(strvalue,"(i1)", iostat=ierr) iplatform
+        if(ierr .ne. 0) then
+            stop 'Error: Cannot convert FORTCL_PLATFORM value into an integer.'
+        endif
+        if(iplatform > num_platforms) then
```

Comment:
Just to idiot proof it, please include a check that `iplatform` is > 0 or -1 (whichever is correct).

Reply:
Done

Reply:
Thanks for improving the message. Please could you check for `iplatform <= 0` rather than just `==0`.

Reply:
Done